Lib/traceback.py

```diff
@@ -520,6 +530,39 @@ def _byte_offset_to_character_offset(str, offset):
     return len(as_utf8[:offset + 1].decode("utf-8"))
 
 
+def _extract_caret_anchors_from_line_segment(segment):
+    import ast
+
+    try:
+        tree = ast.parse(segment)
+    except SyntaxError:
+        return None
+
+    if len(tree.body) != 1:
+        return None
+
+    statement = tree.body[0]
+    match statement:
+        case ast.Expr(expr):
+            match expr:
+                case ast.BinOp():
+                    operator_str = segment[expr.left.end_col_offset:expr.right.col_offset]
```

Comment:
We should probably have a try catch around these potentially erroring statements, otherwise if the file changes underneath between an error being thrown the traceback machinery could throw exceptions internally.

Reply:
We probably need to do it in the top-level, just like the C implementation.